src/GitHub.App/Services/UsageTracker.cs

```diff
@@ -0,0 +1,378 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Reactive;
+using System.Reactive.Linq;
+using Akavache;
+using GitHub.Caches;
+using GitHub.Extensions.Reactive;
+using GitHub.Models;
+using GitHub.Settings;
+using ReactiveUI;
+using Rothko;
+
+#pragma warning disable CS0649
+
+namespace GitHub.Services
+{
+    [Export(typeof(IUsageTracker))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class UsageTracker : IUsageTracker
+    {
+        // Whenever you add a counter make sure it gets added to _both_
+        // BuildUsageModel and ClearCounters
+        const string GHLastSubmissionKey = "GHLastSubmission";
+        const string GHCreateCountKey = "GHCreateCountKey";
+        const string GHCloneCountKey = "GHCloneCount";
+        const string GHPublishCountKey = "GHPublishCountKey";
+        const string GHCreateGistCountKey = "GHCreateGistCountKey";
+        const string GHOpenInGitHubCountKey = "GHOpenInGitHubCountKey";
+        const string GHLinkToGitHubCountKey = "GHLinkToGitHubCountKey";
+        const string GHUpstreamPullRequestCount = "GHUpstreamPullRequestCount";
+        const string GHLoginCountKey = "GHLoginCountKey";
+        const string GHLaunchCountKeyDay = "GHLaunchCountDay";
+        const string GHLaunchCountKeyWeek = "GHLaunchCountWeek";
+        const string GHLaunchCountKeyMonth = "GHLaunchCountMonth";
+
+        static readonly NLog.Logger log = NLog.LogManager.GetCurrentClassLogger();
+
+        readonly IMetricsService client;
+        readonly Lazy<ISharedCache> cache;
+        readonly Lazy<IRepositoryHosts> repositoryHosts;
+        readonly IPackageSettings userSettings;
+
+        [ImportingConstructor]
+        public UsageTracker(
+            Lazy<ISharedCache> cache,
+            Lazy<IRepositoryHosts> repositoryHosts,
+            IPackageSettings userSettings,
+            IUIProvider serviceProvider)
+        {
+            this.cache = cache;
+            this.repositoryHosts = repositoryHosts;
+            this.userSettings = userSettings;
+            this.client = (IMetricsService)serviceProvider.GetService(typeof(IMetricsService));
```

Comment:
This is going to blow up miserably if the service is not found. You need to use `TryGetService<IMetricsService>` here.